src/components/common/query/QueryView.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+
+import { Button, Typography } from '@material-ui/core';
+
+import { Link } from 'react-router-dom';
+import SidebarDrawer from 'components/dataset/data/sidebar/SidebarDrawer';
+import QueryViewDropdown from './QueryViewDropdown';
+import JadeTable from '../../table/JadeTable';
+import SnapshotPopup from '../../snapshot/SnapshotPopup';
+import AppBreadcrumbs from '../../AppBreadcrumbs/AppBreadcrumbs';
+import { BREADCRUMB_TYPE } from '../../../constants';
+
+const styles = (theme) => ({
+  pageRoot: { ...theme.mixins.pageRoot },
+  pageTitle: { ...theme.mixins.pageTitle },
+  wrapper: {
+    paddingTop: theme.spacing(0),
+    padding: theme.spacing(4),
+  },
+  scrollTable: {
+    overflowY: 'auto',
+    height: '100%',
+  },
+});
+
+function QueryView({
+  canLink,
+  classes,
+  handleChangeTable,
+  updateDataOnChange,
+  handleDrawerWidth,
+  panels,
+  queryParams,
+  resourceId,
+  resourceLoaded,
+  resourceName,
+  resourceType,
+  selected,
+  selectedTable,
+  sidebarWidth,
+  tableNames,
+}) {
+  return (
+    <Fragment>
+      {resourceLoaded && (
+        <div className={classes.pageRoot}>
+          <AppBreadcrumbs
+            context={{
+              type:
```

Comment:
I would love to make typescript types for datasets and snapshots. Then, we could just pass the dataset/snapshot to `AppBreadcrumbs` and it could figure out what to do based on the type instead of needing to pass the resource type in as a parameter.